moveit_ros/moveit_servo/include/moveit_servo/utils/common.hpp

```diff
@@ -54,6 +54,23 @@
 
 namespace moveit_servo
 {
+/**
+ * \brief Get the base frame of the current active joint group or subgroup's IK solver.
+ * @param robot_state A pointer to the current robot state.
+ * @param group_name The currently active joint group name.
+ * @return The IK solver base frame, if one exists, otherwise std::nullopt.
+ */
+std::optional<std::string> getIKSolverBaseFrame(const moveit::core::RobotStatePtr& robot_state,
```

Comment:
Can we avoid the pointer here and below and just do `const moveit::core::RobotState &robot_state`?

Reply:
I think the robot state which is extracted from

```
moveit::core::RobotStatePtr CurrentStateMonitor::getCurrentState() const
```

already returns in pointer form, so is it fine to keep as is?

Reply:
That's fine if this is only used in this context. If we ever put it somewhere more general I think we should just take the const&.
`shared_ptr` is viral, it easily spreads everywhere. If this takes a `shared_ptr`, it can't really can't be used unless you have a `shared_ptr` already. Whereas if this takes a `const&`, you can call it directly with a `RobotState` object, but also with a `shared_ptr` or `unique_ptr` by derefencing, so it's more usable.